### ISSUE TITLE
Invalid Static Library Architecture

### ISSUE THREAD

**Issue by gabe-terrell** [RapidApplicationDevelopment/FLIROneSDK-AGC#1]
Hello,

I know this repository is almost a year old, so I'm not sure if it still actively maintained, but currently FLIR [still recommends this library](https://developer.flir.com/developer-faq/) as the only solution to locking the temperature range in iOS applications.

I am trying to link this library into a new application, but I am unable to properly build. I followed [these instructions](https://www.raywenderlich.com/41377/creating-a-static-library-in-ios-tutorial) to link header and binary files, and the project recognizes the binary, but I get the compile-time error that:

`Undefined symbols for architecture armv7:
  "_OBJC_CLASS_$_RosebudAdapter", referenced from:
      objc-class-ref in libFLIROneSDKAGC.a(FLIROneSDKAGC.o)
ld: symbol(s) not found for architecture armv7` 

[Stack Overflow suggests](http://stackoverflow.com/questions/6429494/undefined-symbols-for-architecture-armv7?page=1&tab=votes#tab-top) that this is because the "static library is built for another architecture like i386".

Is it possible to convert this library into a [fat static library](http://stackoverflow.com/questions/3520977/build-fat-static-library-device-simulator-using-xcode-and-sdk-4) or to release the original .m file (as opposed to a binary) pertaining to this project if it is no longer maintained? 

Thanks,
Gabe

**Comment by RapidApplicationDevelopment** [comment]
@gabe-terrell 

Hmm...

The library is a fat file containing armv7, i386, x86_64, and arm64 architectures.

Before we go any further, I just want to make sure that you have the FLIROneSDK properly linked into your project. Were you able to interact with the FLIROneSDK before you added libFLIROneSDKAGC?

i.e., Would the following code compile and link without errors?
`#import <FLIROneSDK/FLIROneSDK.h>`
`  ... `
`[[FLIROneSDKStreamManager sharedInstance] addDelegate:self];`

I ask because symbol `_OBJC_CLASS_$_RosebudAdapter` is defined in the FLIROneSDK and used internally by libFLIROneSDKAGC. If the linker cannot find the FLIROneSDK, then it will not be able to find that symbol.


Let me know,

Clay

**Comment by gabe-terrell** [comment]
Hi Clay,

You are correct, it was an issue linking the FLIROneSDK. 

I am building my primary project in Swift, and was having difficulty getting the bridging-header to recognize the FLIROneSDKAGC header file, so I tried recreating the problem in a new Objective-C app, and evidently I skipped a step in the setup of the FLIR SDK.

Once I fixed that, everything in the Objective-C app worked. Interestingly enough, the bridging-header still wouldn't recognize the FLIROneSDKAGC, but if I added a wrapper Objective-C class, that class would recognize the FLIROneSDKAGC, and in turn the bridging-header recognized the wrapper class. A bit of a goofy solution, but it seems to be working now on my main project.

One concern I still have though is the way `[FLIROneSDKAGC lockAGC]` seems to work. It appears to just lock whichever arbitrary color mapping the camera was using at the time based on the temperatures in the current image, as opposed to just having a purely static color mapping regardless of what time the `lockAGC` call was made. 

Is it possible to guarantee that a specific color mapping will always occur when calling the `lockAGC` function?

Thanks for your time,
Gabe
